### ISSUE TITLE
Proposal for a VF project for the Mutual Aid Network Mapping Summit

### ISSUE THREAD

**Issue by bhaugen** [valueflows/valueflows#135]
@fosterlynn and I just spent a week in Madison WI at the Mutual Aid Network Mapping Summit.
* [overview](https://docs.google.com/document/d/1n1yOZ8UkaXYETFcosBShxGgWzuGdqx-h9u4bEmpdqPA/edit?usp=sharing)
* [videos](https://www.youtube.com/channel/UCBmSlCSPu-cs3GCAkzIe6dA)

At their invitation, we introduced our older regional economic analysis software, also based on the REA ontology, [Locecon](http://locecon.org/). This was part of the [programme I outlined in 2005](http://mikorizal.org/BeyondTheEnterprise.html) for what I planned to work on as I retired.

Locecon takes the VF model and vocabulary to an aggregate level, where Economic Functions represent groups of Economic Agents doing roughly the same kinds of production.

The Mutual Aid Network participants picked up on the idea and even started calling it "value flows". Here's [the value flows model we all co-created during the summit](http://locecon.org/clusters/cluster/25/).

People in Chicago and Asheville North Carolina are also starting to try the software.

The problem is that the software is something that I hacked together for [this project in Nova Scotia](http://locecon.org/nova-story/). It needs a lot of work (I called it "user-hostile"), and is the wrong architecture for the problem.

All of these communities, and all of the agents in these communities, want to be able to create their own economic profiles somewhat independently but have them be findable, shareable, linkable, and combinable in geographic maps as well as resource flow diagrams.

They also want to be able to summarize actual detail economic data to populate the numbers in the aggregate regional-analysis models. Here's [a sketch we wrote for a conversation with the TransforMap people](https://docs.google.com/document/d/1CWaMM76bM4re_TabCyHQOLkqYbP8QxO2ZP3fn4CJOF4/edit?usp=sharing) comparing their model with an OVN model (like Value Flows) and the Locecon model. 

We described this idea in a hand-wavy sort of way at the summit and got enthusiastic and universal approval.

A VF vocab and linked open data could be the perfect architecture.

What do you all think?

**Comment by elf-pavlik** [comment]
Locecon reminds me of https://www.communecter.org/ which @oceatoon works one. Today I will also meet with a friend who develops very similar project https://github.com/ralexrdz/vientos where @ralexrdz does the nodejs dev. They already have in it digital http://needsandresourcesboard.org/

Maybe we could have a call together next week to coordinate our efforts and possibly give current state of VF broader review?

**Comment by bhaugen** [comment]
> Maybe we could have a call together next week to coordinate our efforts and possibly give current state of VF broader review?

Yes please.

**Comment by fosterlynn** [comment]
> Locecon reminds me of https://www.communecter.org/ which @oceatoon works one

It would be great to talk to @oceatoon about it.  We have both tried to get coordinated on it, but not successfully yet.  My impression though, is that Communecter does not deal with resource flows, at least in general?  (Has events - the kind you go to; and projects?)  But still, I'm totally in favor of all attempts at collaboration on this stuff!!!

**Comment by bhaugen** [comment]
If you only looked at the geographical map of that Madison resource flow model and didn't check out the other tabs, you might have missed the difference between the geo map and a resource flow graph. Here's the resource flow view: http://locecon.org/clusters/network/25/

They'd like to overlay the resource flows on the geo map. Among other improvements.

**Comment by bhaugen** [comment]
> They already have in it digital http://needsandresourcesboard.org/

Needs and resources seems like it could be pretty close to what we have in mind, but I couldn't find an example that I could read.  And pdfs are also the wrong architecture, unless I missed something. But the idea seems somewhat similar. The Mutual Aid Networks, though, are trying to organize resource flows between collaborating organizations in a region. They mostly know each other and want to develop a mutual coordination economy among and between them.

**Comment by bhaugen** [comment]
Here's Lynn's model for the simplest possible sets of relationships for a multi-level regional analysis.
![locecon flows](https://cloud.githubusercontent.com/assets/117439/15994252/ee41c0aa-30c4-11e6-912c-3039805c4c33.png)

She can explain anything that is not obvious.

**Comment by bhaugen** [comment]
The resource types on the Function layer could be broader than those on the Person or Group layer.

**Comment by elf-pavlik** [comment]
In you diagram, you have _produces_ and _uses_. Since _produce_ increments resources and _use_ doesn't decrement it. Does it mean that at some point environment will saturate and people can continue to _use_ without need to _produce_ any more? Otherwise I think we may also need some _consumes_ on this diagram.

**Comment by bhaugen** [comment]
Lynn originally had consumes, I changed it to uses because I thought that would be more general. I didn't want to complexify with different event behaviors, because she did this for the Mutual Aid Network audience who are not yet that deep into VF discussions. 

What do you think would be a better general word?

**Comment by bhaugen** [comment]
Also, re those other projects: I really think LOD with WebIds and all might be the fitting architecture for this wannabe project. http://needsandresourcesboard.org/ could probably use it, it would be more usable than pdfs. Communector seems like a platform, where you need to register and participate using their software, inside their walls.better

**Comment by bhaugen** [comment]
Lynn suggests just inputs and outputs.

**Comment by elf-pavlik** [comment]
> Lynn suggests just inputs and outputs.

:+1: which also suggests modelling projects as long run processes with various sub processes. We had chat @fosterlynn discussing defining vf:Process and vf:Agent as disjoint classes, possibly for each agent we could have distinct entity representing its top level process.

related to: https://github.com/valueflows/process/issues/16

> Also, re those other projects: I really think LOD with WebIds and all might be the fitting architecture for this wannabe project.

During http://www.decentralizedweb.net/ @edwardwest had chat with TimBL about https://solidplatform.org/ Let's work on some simple Solid based prototype in next weeks. Myself I would like to make make _talent_ app where people would add skills to their personal profiles, based on wikidata URIs.

**Comment by bhaugen** [comment]
> Let's work on some simple Solid based prototype in next weeks. 

I'm behind, as usual, but committed to start exploring @melvincarvalho 's demos this Wednesday as a way to get some experience with Solid.

> Myself I would like to make make talent app where people would add skills to their personal profiles, based on wikidata URIs.

Semi-related to resource flow graphs.

**Comment by coopchange** [comment]
+1 to Inputs and Outputs.

Not sure if worth noting here, but as we talked about the interface by
which people would provide their information (i.e. InputType, OutputType,
Function, Geo-Location, RelationshipTypes & specific Relationships{that tie
to other Agents}, etc.) as potentially being something similar to the
Poshterity Budgeting exercise run by Stephanie, it occurs to me  the Robbin
Hood Collective (RHC) X Hylo collaboration appears to be coming along, and
is somewhat similar in nature to what we were discussing in the last two
days of the exercise.

That is to say, Hylo is a social media platform that enables users to share
media with the tags: "Requests", "Offers", "Intentions", "Events", &
"Projects"; everyone in this thread thus far has used it, I believe, within
CTA forum.

What they hope to do is embed a kickstarter-like mechanism, similar to
cobudget, but with the capacity to nest sub-buckets (I think); the example
from the video is that someone wants to paint a wall, and has
project-mgmt_skill but doesn't have paint or painting_skill; they propose a
bucket of $600 to paint it, with $300 suggested for agent with
painting_skill and $300 for ResourceType:paint ; then, within the proposal,
someone can communicate: "I have used paint that I'll sell for $150", or "I
would paint for $350"... anyways, this is just something to keep in mind in
terms of interoperability and what is out there, as I understand that this
should be open source.

_More info_: above is informed by what I heard in this [video](https://youtu.be/C6K_WB1sFGw), after Alanna Krause's section & sometime
during Dan Hassan's explanation of what RHC is doing.

Dan & Alanna gave me permission to share the above video into public spaces
with allies & friends, and hope to hear the feedback on it. _The video
details_ some of the Request/Offer cobudgeting features being programmed
into Hylo (I'll share it in the relevant issue that Dan Hassan started).

**Comment by elf-pavlik** [comment]
> > Myself I would like to make make talent app where people would add skills to their personal profiles, based on wikidata URIs.
> 
> Semi-related to resource flow graphs

IMO any _let's make it happen together_ collaboration requires people to contribute skilled work. I plan to use [The Cognitive Characteristics Ontology](http://smiy.sourceforge.net/cco/spec/cognitivecharacteristics.html#cognitive_characteristic) to describe skills and Wikidata IRIs as common references. Also any process offering service to community as its output needs various inputs with skilled work on inputs.

**Comment by bhaugen** [comment]
> IMO any let's make it happen together collaboration requires people to contribute skilled work.

Agreed. Probably shd just have written "related to resource flow graphs".

**Comment by ralexrdz** [comment]
@elf-pavlik I'm up for it too! Let me know proposed dates to see if I'm available ^^

**Comment by elf-pavlik** [comment]
Since you focus on instances of _foaf:Agent_ and not so much on instances of _vf:Process_. I seems that [valueflows/intent](https://github.com/valueflows/intent) comes most relevant here? Where intents need to clearly map to input/output of a process or processes run by that agent.
- _vf:Request_ often will get created based on some planned input
- _vf:Offer_ often will get created based on some existing or planned output (for services pretty much always planned)

I think on the diagram _offers_ & _requests_ would fit well if we focus on agents. While _outputs_ & _inputs_ where we focus on processes.

**Comment by bhaugen** [comment]
> Since you focus on instances of foaf:Agent and not so much on instances of vf:Process. 

Who's "you" in that sentence? 

> I think on the diagram offers & requests would fit well if we focus on agents. While outputs & inputs where we focus on processes.

If me, I wanted to keep the proposed economic profiles really simple. Inputs and outputs might not be the correct word, but I just wanted agents to be able to publish what they produce and what they need on an ongoing basis. When we did that in Madison, a lot of people focused on something they needed right now, as opposed to what they need regularly, over and over again.

Immediate needs are a good basis for requests, and immediate availability is a good basis for offers. But for regional economic analysis, the ongoing resource flow potentials are more important.

**Comment by elf-pavlik** [comment]
I think trying to define _vf:Project_ as ongoing process run by an Agent and having single primary Function would help with drawing distinctions. 
- single Agent can run many Projects
- each Project has primary Function
- Projects have inputs & outputs (ongoing process, which aggregates more specific processes)
- Agents have requests & offers (as intents)

From past conversations I have impression of common tendency to conflate Agent and Process into a Project. It seems helpful to me to keep distinction between:
- Project - ongoing process
- Project Group (Team, Crew ...) - agent running the process

Last but not least, I really appreciate all conversations which attempt to clarify how requests & offers relate to inputs & outputs!

**Comment by bhaugen** [comment]
> From past conversations I have impression of common tendency to conflate Agent and Process into a Project.

Not sure where you detected that tendency. I think @fosterlynn and I have been consistent in NRP where a Project means roughly what Sensorica meant by it, that is, a long-running form of organization run by a project team and in which usually many Processes take place.

In VF, we have also taken in the other meaning of Project that we see in some networks, that of a set of processes with a goal and a deadline, and then the goal is met, the project is finished. 

In other words, Project in the wild has at least those two different meanings, which is why we have hesitated to bake it into the VF vocab with one of them.

> Project - ongoing process

The long-running organization version of Project, as in Sensorica, usually contains many processes. For example:
http://nrp.sensorica.co/accounting/project-wip/326/
http://nrp.sensorica.co/accounting/finished-processes/326/

**Comment by fosterlynn** [comment]
Coming in a bit late here....  :)

I'm not sure Project is a term that makes sense for something like this focused on regional / local economic analysis.  To me it connotes something more about creating something new, whether it is the shorter term goal oriented project or the Sensorica type project.  I think most people involved in Madison won't think of their activities as projects, but more as just what needs to be done ongoing for maintaining life.  Like the coop cab company; or the people who recover extra food from local companies and distribute it; or the swap centers for exchanging stuff.  They are basically looking at re-distributing resources so the neighborhoods that need them can get them.

(Although in VF I am fine considering Project as a vocab term.  And maybe I am wrong about the above, but I did want to explain more how they think of it, irrespective of term.)

Focusing on wants and offers also creates the impression of immediacy, vs long term ongoing economic activity.  (As Bob explained, we made that mistake initially in Madison mapping.)

And yes, we need some new generic terms for Inputs and Outputs.  Produces seems OK; we need something on the input side to cover consumes and uses.

**Comment by coopchange** [comment]
> I think most people involved in Madison won't think of their activities as projects

I agree a bunch with @fosterlynn's comments toward "Project" connotation, and that this may be best framed as _just keep doing what you're already doing_, but I'm not sure about the quote I distilled- I guess I didn't see the word "Project" on the board or the map, particularly, but I've a vague memory of talking with the Food Not Bombs Madison folks (~2 yrs ago) ago in "Project"-terms, for example.

Good take aways re: the _wants/offers_ creating sense of immediacy

> And yes, we need some new generic terms for Inputs and Outputs. Produces seems OK; we need something on the input side to cover consumes and uses.

One thesaurus gave me "utilize" as a synonym for consume; also, I found that "employ" is a synonym for both "use" and of "consume", though I feel like the connotations/denotations are stickier.

"Apply" also comes to mind, as when one "applies" a bandage it is consumed, but when one "applies" a crowbar, it is not.

**Comment by fosterlynn** [comment]
@coopchange thanks for researching the terms!  Nothing is jumping out at me as being ultra-clear.

I'm kind of migrating back to "consumes" at least as a placeholder, although it might create confusion.  If you think of it as consuming time (usage of equipment/space or human work), it fits better.  (I'm working on the LocEcon UI, and am just leaving "consumes" for now....  We can continue to think in background.....)

**Comment by djodjoni** [comment]
:+1:  :)
